packages/bvaughn-architecture-demo/components/console/MessagesList.tsx

```diff
@@ -80,44 +83,49 @@ function MessagesList({ forwardedRef }: { forwardedRef: ForwardedRef<HTMLElement
       listItems.push(currentTimeIndicator);
     }
 
-    if (isEventLog(loggable)) {
-      listItems.push(
-        <EventLogRenderer
-          key={index}
-          index={index}
-          isFocused={loggable === currentSearchResult}
-          eventLog={loggable}
-        />
-      );
-    } else if (isPointInstance(loggable)) {
-      listItems.push(
-        <LogPointRenderer
-          key={index}
-          index={index}
-          isFocused={loggable === currentSearchResult}
-          logPointInstance={loggable}
-        />
-      );
-    } else if (isProtocolMessage(loggable)) {
-      listItems.push(
-        <MessageRenderer
-          key={index}
-          index={index}
-          isFocused={loggable === currentSearchResult}
-          message={loggable}
-        />
-      );
-    } else if (isTerminalExpression(loggable)) {
-      listItems.push(
-        <TerminalExpressionRenderer
-          key={index}
-          index={index}
-          isFocused={loggable === currentSearchResult}
-          terminalExpression={loggable}
-        />
-      );
-    } else {
-      throw Error("Unsupported loggable type");
+    const isLoaded =
+      loadedRegions !== null &&
+      isPointInRegions(getLoggableExecutionPoint(loggable), loadedRegions.loaded);
+    if (isLoaded) {
```

Comment:
The new Console will deviate from the old Console here, by hiding loggables that aren't in loaded regions. This means that for longer recordings, the new Console _may_ render fewer things than the old Console.